### PR TITLE
Inflatable Walls and Doors

### PR DESCRIPTION
**Pull request by ike709**
:cl: ike709
add: Inflatable walls and doors are now a thing. They're great bandaid fixes for hull breaches.
add: Engineers now start with a box of inflatable walls and doors. More can be acquired in an Engi-Vend.
/:cl:

image

I ported these from paradise. Just looked at the code, no specific PR.
Credit to whoever made them originally, I have no clue who you are.

### CODE REVIEW

File: code/game/objects/structures/inflatable.dm [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+	return
+
+/obj/structure/inflatable/proc/hit(var/damage, var/sound_effect = 1)
+	health = max(0, health - damage)
+	if(sound_effect)
+		playsound(loc, 'sound/effects/Glasshit.ogg', 75, 1)
+	if(health <= 0)
+		deflate(1)
+
+/obj/structure/inflatable/AltClick()
```
Comment (Vivalas) [inline code/game/objects/structures/inflatable.dm]: I don't know if this is a good idea I just prefer the right-click verb, makes for less ambiguous accidents and people claiming misclicks.

File: code/game/objects/structures/inflatable.dm [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+	anchored = 1
+	opacity = 0
+
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "door_closed"
+
+	var/state = 0 //closed, 1 == open
+	var/isSwitchingStates = 0
+
+/obj/structure/inflatable/door/attack_ai(mob/user as mob) //those aren't machinery, they're just big fucking slabs of a mineral
```
Comment (Vivalas) [inline code/game/objects/structures/inflatable.dm]: kek this copypasta comment

File: code/game/objects/structures/inflatable.dm [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+		return
+	else if(isrobot(user)) //but cyborgs can
+		if(get_dist(user,src) <= 1) //not remotely though
+			return TryToSwitchState(user)
+
+/obj/structure/inflatable/door/attack_hand(mob/user as mob)
+	return TryToSwitchState(user)
+
+/obj/structure/inflatable/door/CanPass(atom/movable/mover, turf/target, height=0)
+	if(istype(mover, /obj/effect/beam))
```
Comment (Vivalas) [inline code/game/objects/structures/inflatable.dm]: But isn't inflatable shit always opacity = 0?

Comment (ike709) [inline code/game/objects/structures/inflatable.dm]: Probably? I changed the return from "return !opacity" to "return 1".

Was that the issue or is the if(istype) check just totally unnecessary?

Comment (Vivalas) [inline code/game/objects/structures/inflatable.dm]: It's unnecessary

**Pull request merged**